source/common/upstream/logical_dns_cluster.cc

```diff
@@ -46,10 +52,16 @@ LogicalDnsCluster::LogicalDnsCluster(const envoy::api::v2::Cluster& cluster,
     NOT_REACHED;
   }
 
-  const auto& socket_address = hosts[0].socket_address();
+  const auto& locality_lb_endpoint = locality_lb_endpoints[0];
+  const auto& lb_endpoint = locality_lb_endpoint.lb_endpoints()[0];
+
+  const envoy::api::v2::endpoint::Endpoint& endpoint = lb_endpoint.endpoint();
+  const envoy::api::v2::core::SocketAddress& socket_address = endpoint.address().socket_address();
   dns_url_ = fmt::format("tcp://{}:{}", socket_address.address(), socket_address.port_value());
   hostname_ = Network::Utility::hostFromTcpUrl(dns_url_);
   Network::Utility::portFromTcpUrl(dns_url_);
+  priority_state_manager_->initializePerPriority(locality_lb_endpoint);
+  context_ = std::make_shared<ResolveTargetContext>(locality_lb_endpoint, lb_endpoint);
```

Comment:
I'm a little confused about why we need context_. Isn't locality_lb_endpoint and lb_endpoint effectively constant? Can't we just refer to them directly as constance references?